### DISCUSSION TITLE
Is there way to implicitly start the fetch process rather than automatically?

### DISCUSSION THREAD

**Discussion by jaequery** [TanStack/query#1356] [Q&A]
Let say we have: 

```
const {
   data,
   error,
   isLoading
 } = useQuery('profileData', () =>
   fetch('https://api.github.com/users/SoorajSNBlaze333')
   .then(res => res.json()))
```

This will apparently fetch the data as soon as the component is rendered. But let's say I only want to fetch on a specific event. Is there something like this I can do?

```
const {
   fetch,
   data,
   error,
   isLoading
 } = useQuery('profileData', () =>
   fetch('https://api.github.com/users/SoorajSNBlaze333')
   .then(res => res.json()), { autoFetch: false ))
```

This way, on later parts of code i can just call fetch() wherever appropriate

**Comment by TkDodo** [comment]
`useQuery` also returns a method called `refetch` that you can execute when you need to. But i have barely ever needed it - I prefer changing the queryKey to „trigger“ re-ferches.

**Comment by sergiodxa** [comment]
You can set the option enabled to false I totally and change it to true after your action, this will only trigger the fetch when enabled is true
